app/models/direct_file_data.rb

```diff
@@ -317,22 +317,25 @@ def fed_total_adjustments=(value)
     write_df_xml_value(__method__, value)
   end
 
-  def total_state_tax_withheld
-    total = 0
-    parsed_xml.css('IRSW2').map do |w2|
-      amt = w2.at('StateIncomeTaxAmt')&.text.to_i
-      total += amt
+  def total_1099r_state_tax_withheld
+    selector = Df1099rAccessor.selectors[:StateTaxWithheldAmt]
+    form1099r_nodes.sum do |f1099r|
+      f1099r.at(selector)&.text.to_i
     end
-    total
   end
 
-  def total_local_tax_withheld
-    total = 0
-    parsed_xml.css('IRSW2').map do |w2|
-      amt = w2.at('LocalIncomeTaxAmt')&.text.to_i
-      total += amt
+  def total_w2_state_tax_withheld
+    selector = DfW2Accessor.selectors[:StateIncomeTaxAmt]
+    w2_nodes.sum do |w2|
+      w2.at(selector)&.text.to_i
+    end
+  end
+
+  def total_w2_local_tax_withheld
+    selector = DfW2Accessor.selectors[:LocalIncomeTaxAmt]
+    w2_nodes.sum do |w2|
+      w2.at(selector)&.text.to_i
```

Comment:
Sorry for not catching this while we were pairing, but if you want to get rid of the direct access on `selectors`, you could replace this loop with (I think) `w2s.sum(&:LocalIncomeTaxAmt)`

Reply:
And similarly for the other methods just above this one.

Reply:
:chefs-kiss: